aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -409,20 +414,34 @@ private String getIdentityValue(String identityClaim) throws ApiException {
         return groups;
     }
 
-    private CognitoUserPoolsAuthProvider getAuthProvider() throws ApiException {
-        CognitoUserPoolsAuthProvider cognitoProvider = authProvider.getCognitoUserPoolsAuthProvider();
-        if (cognitoProvider == null) {
-            try {
-                cognitoProvider = new DefaultCognitoUserPoolsAuthProvider();
-            } catch (ApiException exception) {
+    private String getAuthToken(AuthorizationType authType) throws ApiException {
+        final String token;
+        switch (authType) {
+            case AMAZON_COGNITO_USER_POOLS:
+                CognitoUserPoolsAuthProvider cognitoProvider = authProvider.getCognitoUserPoolsAuthProvider();
+                if (cognitoProvider == null) {
+                    cognitoProvider = new DefaultCognitoUserPoolsAuthProvider();
+                }
+                return cognitoProvider.getLatestAuthToken();
+            case OPENID_CONNECT:
+                OidcAuthProvider oidcProvider = authProvider.getOidcAuthProvider();
+                if (oidcProvider == null) {
+                    throw new ApiException(
+                        "OidcAuthProvider interface is not implemented.",
+                        "Configure AWSApiPlugin with ApiAuthProviders containing an implementation of " +
+                            "OidcAuthProvider interface that can vend a valid JWT token."
+                    );
+                }
+                return oidcProvider.getLatestAuthToken();
+            case API_KEY:
+            case AWS_IAM:
+            case NONE:
+            default:
                 throw new ApiException(
-                        "Attempted to subscribe to a model with owner based authorization without a Cognito provider",
-                        "Did you add the AWSCognitoAuthPlugin to Amplify before configuring it?"
+                    "Failed to obtain access token from the configured auth provider.",
+                    "Verify that the API is configured with either Cognito User Pools or OpenID Connect."
```

Comment:
I think this suggested recovery method could be clearer - in the most common cause of this error (forgetting to add the auth plugin) the API could be configured with the auth provider. Suggest leaving that suggestion in there and adding to it what would be a common error if it was OIDC.